packages/xrpl/src/models/transactions/common.ts

```diff
@@ -696,3 +733,170 @@ const _DOMAIN_ID_LENGTH = 64
 export function isDomainID(domainID: unknown): domainID is string {
   return isString(domainID) && domainID.length === _DOMAIN_ID_LENGTH
 }
+
+/* eslint-disable max-lines-per-function -- Required here as structure validation is verbose. */
+/* eslint-disable max-statements -- Required here as structure validation is verbose. */
+
+/**
+ * Validates if MPTokenMetadata adheres to XLS-89d standard.
+ *
+ * @param input - Hex encoded MPTokenMetadata.
+ * @returns Validation error messages and boolean denoting if MPTokenMetadata adheres to XLS-89d standard.
+ */
+export function validateMPTokenMetadata(input: string): {
+  isValid: boolean
+  validationMessages: string[]
+} {
+  const validationMessages: string[] = []
+
+  if (!isHex(input) || input.length / 2 > MAX_MPT_META_BYTE_LENGTH) {
+    validationMessages.push(
+      `MPTokenMetadata must be in hex format and max ${MAX_MPT_META_BYTE_LENGTH} bytes.`,
+    )
+    return { validationMessages, isValid: false }
+  }
+
+  let jsonMetaData: unknown
+
+  try {
+    jsonMetaData = JSON.parse(hexToString(input))
+  } catch (err) {
+    validationMessages.push(
+      `MPTokenMetadata is not properly formatted as JSON - ${String(err)}`,
+    )
+    return { validationMessages, isValid: false }
+  }
+
+  if (
+    !(
+      jsonMetaData != null &&
+      typeof jsonMetaData === 'object' &&
+      !Array.isArray(jsonMetaData)
+    )
```

Comment:
```suggestion
    jsonMetaData == null ||
    typeof jsonMetaData !== 'object' ||
    Array.isArray(jsonMetaData)
```
Simplified and easy to read.